### PR TITLE
Gdal3

### LINKED ISSUE #604: GPS does not work with EPSG:27700 British National Grid
GPS position lat/long is not projected to eastings/northings

### Reproduction Steps:

1. Create project in EPSG:27700
2. Transfer project to Android device
3. Activate GPS
4. Run QField
5. Activate GPS button

### Expected behavior:

Map to centre to GPS position

### Observed behavior:

Map is centred to easting = long, northing = lat. 

### Screenshots and GIFs

Screenshot_20190709-115200

**QField version:** 1.1.0 (Matterhorn 10100993)

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: Have only tried this one version.
* Problem can be reliably reproduced, doesn't happen randomly: Yes
* Problem happens with all files and projects, not only some files or projects: GPS works with a EPSG:3857 project.

### LINKED ISSUE #1072: GPS to wrong location - EPSG: 7855
QFIELD: 1.5.3
QGIS: 3.12
Tested both using qfield sync and without

The GPS marker is not showing true location within QField while within GDA2020 Zone 55 (EPSG: 7855).

The GPS shows location as close to west africa when I am actually within Melbourne, Australia.

I am happy to support implementation of this datum with GPS location support ASAP.
 
Screenshot_20200620-121917_QField

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes #1072
Fixes #604

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-a3416d8800e4f83c3cce76241cd94c0c9bcd51a3-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-a3416d8800e4f83c3cce76241cd94c0c9bcd51a3-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-a3416d8800e4f83c3cce76241cd94c0c9bcd51a3-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-a3416d8800e4f83c3cce76241cd94c0c9bcd51a3-x86_64.apk)

File: sdk.conf [hunk 1]
```diff
@@ -1 +1 @@
-osgeo4a_version=20200611
+osgeo4a_version=gdal
```
Comment (m-kuhn) [inline sdk.conf:1]:
```suggestion
osgeo4a_version=gdal

```

Create a fake commit please

File: CMakeLists.txt [hunk 2]
```diff
@@ -77,6 +77,7 @@ IF (ANDROID)
     ${OSGEO4A_LIB_DIR}/libopenjp2.so
     ${OSGEO4A_LIB_DIR}/libfreexl.so
     ${OSGEO4A_LIB_DIR}/libtiff.so
+    ${OSGEO4A_LIB_DIR}/libpng.so
```
Comment (m-kuhn) [inline CMakeLists.txt:80]:
```suggestion
    ${OSGEO4A_LIB_DIR}/libpng16.so
```

File: qfield.pri [hunk 3]
```diff
@@ -32,6 +32,7 @@ android {
     $${OSGEO4A_STAGE_DIR}/$$ANDROID_TARGET_ARCH$$/lib/libopenjp2.so \
     $${OSGEO4A_STAGE_DIR}/$$ANDROID_TARGET_ARCH$$/lib/libfreexl.so \
     $${OSGEO4A_STAGE_DIR}/$$ANDROID_TARGET_ARCH$$/lib/libtiff.so \
+    $${OSGEO4A_STAGE_DIR}/$$ANDROID_TARGET_ARCH$$/lib/libpng.so \
```
Comment (m-kuhn) [inline qfield.pri:35]:
```suggestion
    $${OSGEO4A_STAGE_DIR}/$$ANDROID_TARGET_ARCH$$/lib/libpng16.so \
```

File: sdk.conf (hunk 1 above)
Comment (m-kuhn) [inline sdk.conf:1]:
```suggestion
osgeo4a_version=20200624
```

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-b2a4a5c63680fdd14b6eb9910b045cdb857a7236-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-b2a4a5c63680fdd14b6eb9910b045cdb857a7236-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-b2a4a5c63680fdd14b6eb9910b045cdb857a7236-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1078-b2a4a5c63680fdd14b6eb9910b045cdb857a7236-x86_64.apk)

**Pull request merged**